splice_machine/src/test/java/com/splicemachine/derby/utils/ShowCreateTableIT.java

```diff
@@ -101,6 +102,26 @@ public static void cleanup()
         classWatcher.closeAll();
     }
 
+    static File tempDir;
+
+    @BeforeClass
+    public static void createTempDirectory() throws Exception
```

Comment:
Can you add these methods to SpliceUnitTest ?

If you don't want to create a temp directory for every class even if it doesn't use it we can create the directory on demand in getExternalResourceDirectory() (if not already created) and remove it if exists in deleteTempDirectory

Reply:
I know this looks like copied code but putting this in SpliceUnitTest would prevent ExternalTableIT, ExternalTablePartitionIT to run parallel:

Since **static File tempDir** is static, it is shared by all instantiations of SpliceUnitTest, especially between ExternalTableIT, ExternalTablePartitionIT, etc. . However, @AfterClass is called after ExternalTableIT is done, then would delete the (shared) tempDir, which might be still used by ExternalTablePartitionIT when running in parallel.

So **static File tempDir** can be static, but static in ExternalTableIT, so that it is not shared with the one in ExternalTablePartitionIT . Another solution would be to have a TestWatcher.

Reply:
Hadn't thought of that, makes sense. Thanks!